src/server/api.rs

```diff
@@ -0,0 +1,102 @@
+use std::{net::ToSocketAddrs};
+use tracing::info;
+use anyhow::Result;
+
+use crate::config;
+use tonic::{transport::Server, Request, Response, Status, Streaming};
+
+mod transaction {
+    tonic::include_proto!("transaction");
+}
+mod account {
+    tonic::include_proto!("account");
+}
+mod trees {
+    tonic::include_proto!("trees");
+}
+mod note {
+    tonic::include_proto!("note");
+}
+mod block {
+    tonic::include_proto!("block");
+}
+mod digest {
+    tonic::include_proto!("digest");
+}
+mod rpc {
+    tonic::include_proto!("rpc");
+}
+
+
+#[derive(Default)]
+pub struct RpcApi {}
+
+#[tonic::async_trait]
+impl rpc::api_server::Api for RpcApi {
+    type CheckNullifiersStream = Streaming<rpc::ResponseBytes>;
+    type ListNotesByTagStream = Streaming<rpc::ResponseBytes>;
+
+    async fn get_block_header(
+        &self,
+        _request: Request<rpc::GetBlockHeaderRequest>,
+    ) -> Result<Response<rpc::ResponseBytes>, Status>
+    {
+        todo!()
+    }
+
+    async fn get_note(
+        &self,
+        _request: Request<rpc::GetNoteRequest>,
+    ) -> Result<Response<rpc::ResponseBytes>, Status>
+    {
+        todo!()
+    }
+
+    async fn check_nullifiers(
+        &self,
+        _request: Request<Streaming<digest::Digest>>,
+    ) -> Result<Response<Self::CheckNullifiersStream>, Status>
+    {
+        todo!()
+    }
+
+    async fn get_transaction_data(
+        &self,
+        _request: Request<rpc::GetTransactionDataRequest>,
+    ) -> Result<Response<rpc::ResponseBytes>, Status>
+    {
+        todo!()
+    }
+
+    async fn list_notes_by_tag(
+        &self,
+        _request: Request<rpc::ListNotesByTagRequest>,
+    ) -> Result<Response<Self::ListNotesByTagStream>, Status>
+    {
+        todo!()
+    }
+
+    async fn submit_transaction(
+        &self,
+        _request: Request<rpc::SubmitTransactionRequest>,
+    ) -> Result<Response<rpc::ResponseBytes>, Status>
+    {
+        todo!()
+    }
+}
+
+pub async fn serve(config: config::Config) -> Result<()> {
+    let host_port = (config.endpoint.host.as_ref(), config.endpoint.port);
+    let addrs: Vec<_> = host_port.to_socket_addrs()?.collect();
+
+    info!(
+        host = config.endpoint.host,
+        port = config.endpoint.port,
+        "Server initializer",
+    );
```

Comment:
This is from the `tracing` crate, output looks like:

```
2023-07-12T17:12:00.804316Z  INFO main miden_node::server::api: src/server/api.rs:92: Server initializer host="localhost" port=57291
```

The library also supports JSON, which later will be great to debug, and the open telemetry.